### PR TITLE
Error and logging improvements

### PR DESCRIPTION
**Pull request by ps2**
Update errors to discern between unknown/invalid data and no data, and also to detect crosstalk. Tracking name of pump sessions for debugging as well. Still logging directly to nslog.  Changes for delegating that is future work

### CODE REVIEW

File: RileyLinkBLEKit/RileyLinkBLEDevice.h [hunk 1]
```diff
@@ -90,7 +90,7 @@ typedef NS_ENUM(NSUInteger, SubgRfspyVersionState) {
 
 - (void) connectionStateDidChange:(nullable NSError *)error;
 
-- (void) runSession:(void (^ _Nonnull)(RileyLinkCmdSession* _Nonnull))proc;
+- (void) runSession:(nonnull NSString*)name proc:(void (^ _Nonnull)(RileyLinkCmdSession* _Nonnull))proc;
```
Comment (loudnate) [inline RileyLinkBLEKit/RileyLinkBLEDevice.h]: `runSessionWithName:(`?

Comment (loudnate) [reply RileyLinkBLEKit/RileyLinkBLEDevice.h]: `usingBlock:` is probably the most cocoa-esque second arg name here too.

Comment (ps2) [reply RileyLinkBLEKit/RileyLinkBLEDevice.h]: Yeah, I think those are improvements.  I really want to port this all to swift. But time.

Comment (loudnate) [reply RileyLinkBLEKit/RileyLinkBLEDevice.h]: well at least the Swift portions have to get upgraded next week anyhow... I can do some renaming then!

**Comment by loudnate** [comment]
LGTM. I don't think there's any need to worry about delegating logging vs migrating to the iOS 10 logging framework.

File: RileyLinkBLEKit/RileyLinkBLEDevice.h (hunk 1 above)
Comment (ps2) [reply RileyLinkBLEKit/RileyLinkBLEDevice.h]: Why are they getting upgraded next week?

Comment (loudnate) [reply RileyLinkBLEKit/RileyLinkBLEDevice.h]: It'll be time to migrate to Swift 3.

Comment (loudnate) [reply RileyLinkBLEKit/RileyLinkBLEDevice.h]: (and all the Objective-C name import logic has changed)

**Pull request merged**